Keen/ContractResolvers/SnakeCaseContractResolver.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace Keen.Core.ContractResolvers
+{
+    using Newtonsoft.Json.Serialization;
+
+    /* 
+     * Found on GithubGist https://gist.github.com/crallen/9238178
+     * 
+     * Credit to Chris Allen https://github.com/crallen 
+     */
+    public class SnakeCaseContractResolver : DefaultContractResolver
```

Comment:
Is this necessary because the version of Json.NET we're using doesn't yet have the `SnakeCaseNamingStrategy` or is it the case that even that strategy does not satisfy your needs? How does this `SnakeCaseContractResolver` handle `camelCase` or `PascalCase` or other casing like `kebab-case`, and how does that compare to Json.NET's own `SnakeCaseNamingStrategy`? How does this treat names that have a `JsonPropertyAttribute` explicitly specifying the name?

http://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_Serialization_SnakeCaseNamingStrategy.htm
http://www.phrohdoh.com/blog/2016/09/02/newtonsoft-json-custom-property-names/